src/symbolize/gimli.rs

```diff
@@ -372,28 +372,31 @@ cfg_if::cfg_if! {
         fn native_libraries() -> Vec<Library> {
             let mut ret = Vec::new();
             unsafe {
-                libc::dl_iterate_phdr(Some(callback), &mut ret as *mut _ as *mut _);
+                libc::dl_iterate_phdr(Some(callback), &mut ret as *mut Vec<_> as *mut _);
             }
             return ret;
         }
 
+        // SAFETY: `info` and `vec` should be valid pointers
```

Comment:
I'm not personally a huge fan of these comments because the "safety" here encompasses far more than what's mentioned here. I think it would be fine to document this without the "SAFETY" header which looks like it's trying to be exhaustive.